ic-cdk/src/api/call.rs

```diff
@@ -110,40 +110,49 @@ impl<T: AsRef<[u8]>> Future for CallFuture<T> {
             } => {
                 let callee = id.as_slice();
                 let args = arg.as_ref();
-                let state_ptr = Arc::into_raw(Arc::clone(&self_ref.state));
+                let state_ptr =
+                    Arc::<RwLock<CallFutureState<T>>>::into_raw(Arc::clone(&self_ref.state));
```

Comment:
The type is written more explicitly. I saw the `callback` and `cleanup` functions' comment also mentions this type. Is this intended to check that `self_ref.state` has the correct type?